### PR TITLE
Add refactoring

### PR DESCRIPTION
**Pull request by mlan080**
#32 

Q: So with original, it returns a hash and we're trying to return a shop instance right? Why is there an error in the irb?

### CODE REVIEW

File: app/models/shop.rb [hunk 1]
```diff
@@ -26,7 +26,8 @@ def self.last
   end
 
   def self.all
-    DB[:shops].all
+    all_records = DB[:shops].all
+    Shop.new(all records)
```
Comment (mlan080) [inline app/models/shop.rb]: It returns an array of hashes but trying to return an array of shops

File: app/models/shop.rb [hunk 2]
```diff
@@ -26,7 +26,9 @@ def self.last
   end
 
   def self.all
-    DB[:shops].all
+    #DB[:shops].all
+    all_records = DB[:shops].all
+    all_records.map{|x| Shop.new(x)}
```
Comment (lucasrenan) [inline app/models/shop.rb]: cool, this is the idea.
maybe we can do it in a single line, like:

```ruby
DB[:shops].all.map{ |attrs| Shop.new(attrs) }
```

File: app/models/shop.rb [hunk 3]
```diff
@@ -26,7 +26,9 @@ def self.last
   end
 
   def self.all
-    DB[:shops].all
+    #DB[:shops].all
```
Comment (lucasrenan) [inline app/models/shop.rb]: leftover

File: spec/shop_spec.rb [hunk 4]
```diff
@@ -42,10 +42,14 @@
   end
 
   describe ".all" do
-    let(:rows) { DB[:shops].all }
+    #let(:rows) { DB[:shops].all }
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: leftover

File: spec/shop_spec.rb [hunk 5]
```diff
@@ -42,10 +42,14 @@
...
+    let!(:first_shop) { Shop.new({name: 'Mando', description: 'is hot stuff'}).create }
+    let!(:second_shop) { Shop.new({name: 'Mandaline', description: 'is hotter stuff'}).create }
 
     it 'returns all rows from the shops table' do
-      expect(Shop.all).to eq(rows)
+      got = Shop.all.map{|x| x.id}
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: this one could be:

```ruby
all_shops = Shop.all.map(&:id)

expect(all_shops).to eq([first_shop, second_shop])
```

File: app/models/shop.rb (hunk 2 above)
Comment (mlan080) [reply app/models/shop.rb]: So attrs is short for attributes?

**Pull request merged**